programmers/workout-clothes.js

```diff
@@ -0,0 +1,20 @@
+function solution(n, lost, reserve) {
+    const differReverse = reserve.filter(x => !lost.includes(x));
+    const differLost = lost.filter(x => !reserve.includes(x));  // 여분의 옷이 있었지만 도둑맞아 여분의 옷이 사라진 사람 제거
+    differLost.sort((a,b) => { 
+        return a-b
+    });
+    for (let i = 0 ; i < differLost.length ; i++) {
+        if (differReverse.indexOf(differLost[i]-1) !== -1) {
+        } else if (differReverse.indexOf(differLost[i]+1) !== -1) {
+            differReverse.splice(differReverse.indexOf(differLost[i]+1), 1);
+        } else {
+            n -= 1;
+        }
+    }
+    return n;
+}
+let n = 5;
+const lost = [4, 2];
+const reserve = [3, 5];
+solution(n, lost, reserve);
```

Comment:
요렇게 forEach로 풀 수도 있습니다.
이제 굉장히 잘 푸시네요~
```suggestion
function solution(n, lost, reserve) {
    const noDupleLost = lost.filter(student => reserve.indexOf(student) === -1);
    const noDupleReserve = reserve.filter(student => (lost.indexOf(student) === -1));

    let answer = n - noDupleLost.length;
    noDupleLost.sort((a, b) => a - b);
    
    noDupleLost.forEach(lostStudent => {
    if (noDupleReserve.indexOf(lostStudent - 1) >= 0) answer ++
    else if (noDupleReserve.indexOf(lostStudent + 1) >= 0) {
        answer++
        let reserveStudent = noDupleReserve.indexOf(lostStudent + 1);
        noDupleReserve.splice(reserveStudent, 1);
    }
  })
  return answer;
}
```